### PR TITLE
feat(console): add SSO connector "connection" tab

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

add SSO connector "connection" tab, including following changes:
1. [console] Refactor the form of SAML and OIDC implemented in SSO connector "guides" so that we can reuse these forms.
 - In guide page, we do not check whether the submitted `config` is complete but in details page we need to do such checks, hence the switch (in this PR, `isFieldCheckRequired` is added) should be added.
 - For `metadataUrl` and `metadata` of SAML connector as well as for OIDC connector, parsed config preview and inline notification should be added.
2. [console] Add metadata/OIDC configuration preview to OIDC/SAML config form to show whether the metadata parsing is successful.
3. [core] Add `shouldReplace` query parameters to API `PATCH /sso-connectors/:id/config` to specify whether should merge the config (in HTTP request payload) with existing config form DB. (for admin console use, we followed the "replace" scheme). Corresponding integration tests are added.

TODO:
- Use text file uploader components to replace multi-line text input.
- Add certificate validator and parser to get its expiration time. Should block save of invalid certificate.

## Testing

Tested locally, programmatic console integration testing will come in upcoming PRs.
image
image
image
image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests (API integration test, will add console integration test in upcoming PRs)
- [x] necessary TSDoc comments (front-end components use normal comments)

### CODE REVIEW

File: packages/console/src/pages/EnterpriseSso/Guide/OidcMetadataForm/ParsedConfigPreview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  providerConfig: ParsedSsoIdentityProviderConfig<SsoProviderName.OIDC>;
+};
+
+function ParsedConfigPreview({ providerConfig }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { authorizationEndpoint, tokenEndpoint, userinfoEndpoint, jwksUri, issuer } =
+    providerConfig;
+  return (
+    <div className={styles.container}>
+      <div>
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSso/Guide/OidcMetadataForm/ParsedConfigPreview/index.tsx:18]: What does the extra `<div>` do? Seems unnecessary

File: packages/console/src/pages/EnterpriseSso/index.tsx [hunk 2]
```diff
@@ -199,7 +200,10 @@ function EnterpriseSsoConnectors() {
             connectorForGuide && (
               <Guide
                 isOpen={Boolean(pathname.endsWith(buildGuidePathname(connectorForGuide.id)))}
-                connector={connectorForGuide}
+                connector={
+                  // eslint-disable-next-line no-restricted-syntax
+                  connectorForGuide as SsoConnectorWithProviderConfigWithGeneric<SsoProviderName>
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSso/index.tsx:205]: Can we avoid this?

File: packages/console/src/pages/EnterpriseSsoDetails/Connection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        onSubmit={onSubmit}
+      >
+        <FormCard
+          title="enterprise_sso_details.upload_idp_metadata_title"
+          description="enterprise_sso_details.upload_idp_metadata_description"
+        >
+          {providerName === SsoProviderName.OIDC ? (
+            // Can not infer the type by narrowing down the value of `providerName`, so we need to cast it.
+            <OidcMetadataForm
+              isGuidePage={false}
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSsoDetails/Connection/index.tsx:92]: Seems `isGuidePage` is optional, so we don't need to pass `false` to it

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/EnterpriseSsoDetails/Connection/index.tsx (hunk 3 above)
Comment (darcyYe) [reply packages/console/src/pages/EnterpriseSsoDetails/Connection/index.tsx:92]: Intentially declare this for better readability to show that this is not on the guide page.

File: packages/core/src/routes/sso-connector/index.ts [hunk 4]
```diff
@@ -248,18 +251,20 @@ export default function singleSignOnRoutes<T extends AuthedRouter>(...args: Rout
...
-        {
-          ...config,
-          ...body,
-        },
+        yes(shouldReplace)
+          ? body
+          : {
+              ...config,
+              ...body,
+            },
```
Comment (simeng-li) [inline packages/core/src/routes/sso-connector/index.ts:267]: replace should call `PATCH ${pathname}/:id` endpoint with { config } body ?

Comment (darcyYe) [reply packages/core/src/routes/sso-connector/index.ts]: Good call, use `PATCH ${pathname}/:id` instead.

**Pull request merged**